packages/aws-cdk-lib/aws-stepfunctions/lib/state-machine.ts

```diff
@@ -460,11 +547,11 @@ export class StateMachine extends StateMachineBase {
       tracingConfiguration: props.tracingEnabled ? this.buildTracingConfiguration() : undefined,
       ...definitionBody.bind(this, this.role, props, graph),
       definitionSubstitutions: props.definitionSubstitutions,
+      encryptionConfiguration: constructEncryptionConfiguration(props, this.defaultPeriodSeconds),
```

Comment:
this will add `encryptionConfiguration` to the outputted state machine no matter what which will be breaking. Should probably be something like 
```suggestion
      encryptionConfiguration: props.enableEncryptedLogging ? constructEncryptionConfiguration(props, this.defaultPeriodSeconds) : undefined
```

Reply:
Thanks for calling this out, I removed the check in previous commits. Will update to: 

```
encryptionConfiguration: props.kmsKey ? constructEncryptionConfiguration(props, this.defaultPeriodSeconds) : undefined,```

Reply:
Update: we're constructing `encryptionConfiguration` like this:

```
encryptionConfiguration: constructEncryptionConfiguration(props.kmsKey, props.kmsDataKeyReusePeriodSeconds),
```

If props.kmsKey is `undefined` we just simply return `undefined`